charts/linkerd-smi/templates/adaptor.yaml

```diff
@@ -21,6 +21,19 @@ spec:
         linkerd.io/extension: smi
         component: smi-adaptor
     spec:
+      {{- with .Values.adaptor.tolerations}}
+      tolerations: 
+{{toYaml . | indent 8}}
+      {{- end }}
+      {{- with .Values.adaptor.nodeSelector }}
+      nodeSelector: 
+{{toYaml . | indent 8}}
+      {{- end }}
+      {{- with .Values.adaptor.affinity.nodeAffinity }}
+      affinity:
+        nodeAffinity:
+{{toYaml . | indent 10 }}
```

Comment:
After having some experience with the affinity fields in the linkerd2 repo, I think the clearest is to have a single free-form `values.adaptor.affinity` setting, perhaps with usage examples in the entry doc in the values.yaml file. That way people can either declare `nodeAffinity` and `podAntiAffinity` at will.